src/main/scala/ai/tripl/arc/ARC.scala

```diff
@@ -423,16 +435,23 @@ object ARC {
       stages match {
         case Nil => None // end
         case (stage, index) :: Nil =>
-          before(stage, index, pipeline.stages)
-          val result = processStage(stage)
-          after(result, stage, index, pipeline.stages)
-          result
-
+          if (runStage(stage, index, pipeline.stages)) {
```

Comment:
I'm not sure this is what we want because if any of the runStage checks return false then none of the before calls on lifecycle plugins will be executed. However I though we wanted the plugins that were allowed to be run - if that's not the case then perhaps we should clarify the expected execution order.

Reply:
This is tricky. Thinking is:
- this is determining whether to run the stage or not.
- if any of the plugins determine the stage should not run then i'm not convinced it should.
- if the stage should not run then logically there is no before/after to consider